python/langsmith/client.py

```diff
@@ -3537,12 +3480,37 @@ def create_examples(
             The ID of the dataset to create the examples in.
         dataset_name : Optional[str], default=None
             The name of the dataset to create the examples in.
+        kwargs: Additional keyword arguments are ignored.
         """
         if dataset_id is None and dataset_name is None:
             raise ValueError("Either dataset_id or dataset_name must be provided.")
-
-        if dataset_id is None:
+        elif dataset_id is None:
             dataset_id = self.read_dataset(dataset_name=dataset_name).id
+        else:
+            pass
+        if attachments:
+            upserts = [
+                ls_schemas.ExampleUpsertWithAttachments(
+                    inputs=in_,
+                    outputs=out_,
+                    attachments=attach_,
+                    dataset_id=cast(uuid.UUID, dataset_id),
+                    metadata=metadata_,
+                    split=split_,
+                    id=id_ or uuid.uuid4(),
+                )
+                for in_, out_, attach_, metadata_, split_, id_ in zip(
+                    inputs,
+                    outputs or [None] * len(inputs),
+                    attachments,
+                    metadata or [None] * len(inputs),
+                    splits or [None] * len(inputs),
+                    ids or [None] * len(inputs),
+                )
```

Comment:
I think this silently cuts off data? Might be good to throw a warning/error if any of inputs/outputs/etc. have different (non-0) lengths, idk